File: src/main/java/org/acme/employeescheduling/dto/GeneralSettings.java
package org.acme.employeescheduling.dto;

import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * General settings associated with a structure (contextual, like Solver Settings).
 * - shiftWindowMode: shift-window granularity in Shift Management ("week" | "month").
 * - autoPopulateFromTemplate: automatically populates empty periods from the location template.
 */
public class GeneralSettings {
    @JsonProperty("id") private int id;
    @JsonProperty("structure_id") private int structureId;
    @JsonProperty("shift_window_mode") private String shiftWindowMode = "month";
    @JsonProperty("auto_populate_from_template") private boolean autoPopulateFromTemplate = false;

    public GeneralSettings() {}

    public GeneralSettings(int id, int structureId, String shiftWindowMode, boolean autoPopulateFromTemplate) {
        this.id = id;
        this.structureId = structureId;
        this.shiftWindowMode = shiftWindowMode;
        this.autoPopulateFromTemplate = autoPopulateFromTemplate;
    }

    public int getId() { return id; }
    public void setId(int id) { this.id = id; }

    public int getStructureId() { return structureId; }
    public void setStructureId(int structureId) { this.structureId = structureId; }

    public String getShiftWindowMode() { return shiftWindowMode; }
    public void setShiftWindowMode(String shiftWindowMode) { this.shiftWindowMode = shiftWindowMode; }

    public boolean isAutoPopulateFromTemplate() { return autoPopulateFromTemplate; }
    public void setAutoPopulateFromTemplate(boolean autoPopulateFromTemplate) { this.autoPopulateFromTemplate = autoPopulateFromTemplate; }
}

File: frontend/src/api/generalSettings.ts
import { api } from './client'

/** Structure-specific general settings (window granularity and automatic population). */
export interface GeneralSettings {
  id: number
  structure_id: number
  /** Shift-window granularity ('week' | 'month'). */
  shift_window_mode: 'week' | 'month'
  /** Automatically populates empty periods from the location template. */
  auto_populate_from_template: boolean
}

export const generalSettingsApi = {
  get: (structureId: number) => api.get<GeneralSettings>(`/general-settings?structureId=${structureId}`),
  save: (structureId: number, value: GeneralSettings) => api.put<GeneralSettings>(`/general-settings?structureId=${structureId}`, value),
}
